backend/modules/metadata_store/base.py

```diff
@@ -18,60 +18,29 @@
 from backend.utils import run_in_executor
 
 
-# TODO (chiragjn): Ideal would be we make `async def a*` abstract methods and drop sync ones
-#   Implementations can then opt to call their sync versions using run_in_executor
 class BaseMetadataStore(ABC):
```

Comment:
All the not implemented methods which are required should be marked with `@abstractmethod`